addons/refuel/functions/fnc_refuel.sqf

```diff
@@ -20,85 +20,89 @@
 
 params [["_unit", objNull, [objNull]], ["_sink", objNull, [objNull]], ["_nozzle", objNull, [objNull]], ["_connectToPoint", [0,0,0], [[]], 3]];
 
-private _config = configFile >> "CfgVehicles" >> typeOf _sink;
-
-private _rate =  getNumber (_config >> QGVAR(flowRate)) * GVAR(rate);
-private _maxFuel = getNumber (_config >> QGVAR(fuelCapacity));
-
-// Fall back to vanilla fuelCapacity value (only air and sea vehicles don't have this defined by default by us)
-// Air and sea vehicles have that value properly defined in liters, unlike ground vehicles which is is formula of (range * tested factor) - different fuel consumption system than ground vehicles
-if (_maxFuel == 0) then {
-    _maxFuel = getNumber (_config >> "fuelCapacity");
+private _rate = if (isNumber (configFile >> "CfgVehicles" >> typeOf _sink >> QGVAR(flowRate))) then {
```

Comment:
I hate this if assignment construct, I'd write it as:
```sqf
private _cfgFlowRate = configFile >> "CfgVehicles" >> typeOf _sink >> QGVAR(flowRate);
private _rate = GVAR(rate); // Default to 1 (jerry cans for example have no flow rate defined)
if (isNumber _cfgFlowRate) then {
    _rate = _rate * getNumber _cfgFlowRate;
};
```
Also less commands due to config stored in variable.

But up to you.